packages/artifact/src/internal/shared/artifact-twirp-client.ts

```diff
@@ -70,6 +70,38 @@ class ArtifactHttpClient implements Rpc {
     }
   }
 
+  /**
+   * Masks the `sig` parameter in a URL and sets it as a secret.
+   * @param url The URL containing the `sig` parameter.
+   * @returns A masked URL where the sig parameter value is replaced with '***' if found,
+   *          or the original URL if no sig parameter is present.
+   */
+  maskSigUrl(url: string): string {
```

Comment:
Instead of having this as a method on clients (therefore needing to export/test the entire client) how about making a singular `maskSignedURLSignature` utility function?

Reply:
Yeah, tried to do that [here](https://github.com/actions/toolkit/pull/1982/commits/944e6b78db75b17247c1e9d1d46f48a3db5237e7#diff-f63fb32fca85d8e177d6400ce078818a4815b80ac7a3319b60d3507354890992), discussed with @Link- and we thought it's okay to have this duplicated code, so reverted the approach with the utility function and put it into each client. Ran into some issue with testing it locally using npm link for some reason. Logically I think it should work with the Utility function in core, so open to trying that out, but hit that issue with the test saying the utility function doesn't exist, despite the compiled code showing it and the typescript compiler not complaining.

Reply:
This is the same conversation we had for cache / artifacts regarding the clients and shared components. It might not be worth it to block this small fix by a larger effort to address the use of a shared library

Reply:
I didn't necessarily mean making it shared across packages, more of a utility function in each. Since it's on the client, we're exporting it now (for tests?). It doesn't need to be on the client class since it it doesn't use any class attributes and could be easier to test _just_ that part.

But this is all nonblocking for the change

Reply:
Ah okay, ya that sounds like a good idea, I can put it in util.ts